bundles/org.openhab.persistence.jpa/src/main/java/org/openhab/persistence/jpa/internal/JpaPersistenceService.java

```diff
@@ -209,10 +209,10 @@ public void store(Item item, @Nullable String alias) {
             Query query = em.createQuery(queryString);
             query.setParameter("itemName", item.getName());
             if (hasBeginDate) {
-                query.setParameter("beginDate", filter.getBeginDate());
+                query.setParameter("beginDate", filter.getBeginDateZoned());
```

Comment:
It looks like support for the java.time API was only [added in JPA 2.2](https://vladmihalcea.com/whats-new-in-jpa-2-2-java-8-date-and-time-types/) whereas the add-on uses OpenJPA 2.4.0 as dependency which is [based](https://openjpa.apache.org/downloads.html) on the JPA 2.0 spec. So the quick fix is probably to convert the ZDTs to Dates here. The nicest fix would be to upgrade OpenJPA to 3.0.x or 3.1.x which is JPA 2.2. compliant, but I don't know how much work that is. :wink:

Reply:
Looks like the POM is using multiple versions: :thinking: 

https://github.com/openhab/openhab-addons/blob/master/bundles/org.openhab.persistence.jpa/pom.xml

Reply:
Hm, I wasn't aware that we have such an old JPA in place here...

> The nicest fix would be to upgrade OpenJPA to 3.0.x or 3.1.x which is JPA 2.2

Hm, do we actually have any active maintainer of this add-on? Or is JPA at all used by people in real life...?

> Looks like the POM is using multiple versions

I think it is only different versions for the lib and the build plugin, isn't it?

Reply:
I did find an occasional user in some community posts using JPA but no active maintainer.

The versions differ but the 2.4.0 version is embedded in the bundle.

Reply:
@wborn I have now changed the code of the `JPAPersistenceService` to use `java.util.Date` internally, so that we should be safe.